Tests/Realm.Tests/Sync/SyncTestBase.cs

```diff
@@ -56,13 +56,13 @@ protected App CreateApp(AppConfiguration config = null)
 
         protected override void CustomTearDown()
         {
-            base.CustomTearDown();
```

Comment:
If a Realm is disposed its sync session should be automatically disposed, so does this manual cleanup have the goal of speeding up the process of memory release or is there more to it?

Reply:
A sync session is independent of the Realm - it may certainly outlive the managed instance. The issue with the order here was that we were calling the base teardown method which disposes and deletes all Realm instances before disposing the session handles. This resulted in race conditions where if a session hasn't been disposed, the deletion would fail.

Reply:
Then I misinterpreted [this comment](https://github.com/realm/realm-dotnet/blob/35b144d92d866a6e42474ec869de4c520c93c140/Realm/Realm/Sync/Session.cs#L28); I assume it simply meant that the user doesn't have to care to manually deallocating it.
Thank you for the explanation.

Reply:
Yes, the automatic lifespan management means that we'll create and terminate sessions based on certain heuristics/internal rules. Typically, this will mean that a session outlives its Realm so that it can upload any remaining changes. This has the unfortunate side effect that deleting a synchronized Realm, even after disposing it, is not exactly safe. This will hopefully be addressed by https://github.com/realm/realm-core/issues/4424.